sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessage.java

```diff
@@ -588,6 +628,34 @@ void setTo(String to) {
      * @see #getViaPartitionKey()
      */
     void setViaPartitionKey(String viaPartitionKey) {
-        this.viaPartitionKey = viaPartitionKey;
+        amqpAnnotatedMessage.getMessageAnnotations().put(VIA_PARTITION_KEY_ANNOTATION_NAME.getValue(), viaPartitionKey);
+    }
+
+    /*
+     * Gets String value from given map and null if key does not exists.
+     */
+    private String getStringValue(Map<String, Object> dataMap, String key) {
+        return dataMap.containsKey(key) ? (String) dataMap.get(key) : null;
```

Comment:
You can just get it instead of doing two map accesses, doesn't work? Same with below.
```java
return (String) dataMap.get(key);
```